libsolidity/parsing/Parser.cpp

```diff
@@ -673,6 +673,8 @@ ASTPointer<EnumDefinition> Parser::parseEnumDefinition()
 	}
 	if (members.empty())
 		parserError(3147_error, "enum with no members is not allowed.");
+	if (members.size() > 256)
+		parserError(1611_error, "enum with more than 256 members is not allowed.");
```

Comment:
I added the check to the parser because we already check for zero members here but maybe it should rather be an error at the analysis stage?

Reply:
Looks like putting it in the parser requires adjusting the grammar. I don't think this change should affect the grammar so I'll move it to the analysis stage after all.

Reply:
`DeclarationTypeChecker` seems like the right place because it also checks e.g. if the arrays have valid lengths, so I moved it there. It's a type error now.